financial-templates-lib/clients/ExpiringMultiPartyClient.js

```diff
@@ -91,25 +91,21 @@ class ExpiringMultiPartyClient {
   }
 
   async update() {
-    // Since this function can have a relatively long run-time, we should be safe and check contract state explicitly at the same block height.
-    // This way we won't experience errors where the contract state changes between calls.
-    this.latestBlock = await this.web3.eth.getBlockNumber();
-
-    this.collateralRequirement = this.toBN(
-      (await this.emp.methods.collateralRequirement().call(undefined, this.latestBlock)).toString()
-    );
-    this.liquidationLiveness = Number(await this.emp.methods.liquidationLiveness().call(undefined, this.latestBlock));
-
-    const events = await this.emp.getPastEvents("NewSponsor", { fromBlock: 0, toBlock: this.latestBlock });
+    const [collateralRequirement, liquidationLiveness, events, cumulativeFeeMultiplier] = await Promise.all([
+      this.emp.methods.collateralRequirement().call(),
+      this.emp.methods.liquidationLiveness().call(),
+      this.emp.getPastEvents("NewSponsor", { fromBlock: 0 }),
+      this.emp.methods.cumulativeFeeMultiplier().call()
+    ]);
+    this.collateralRequirement = this.toBN(collateralRequirement.toString());
+    this.liquidationLiveness = Number(liquidationLiveness);
     this.sponsorAddresses = [...new Set(events.map(e => e.returnValues.sponsor))];
+    this.cumulativeFeeMultiplier = this.toBN(cumulativeFeeMultiplier.toString());
 
     // Fetch information about each sponsor.
     const positions = await Promise.all(
       this.sponsorAddresses.map(address => this.emp.methods.positions(address).call(undefined, this.latestBlock))
```

Comment:
nit: shouldn't we remove latestBlock from this queries (and those below)? Is it even defined anymore?

Reply:
Yeah I just noticed this and made the change